aldryn_search/search_indexes.py

```diff
@@ -43,8 +43,49 @@ def get_plugin_queryset(self, language):
         return queryset
 
     def get_search_data(self, obj, language, request):
+        """
+        In the project settings set up the variable
+
+        PLACEHOLDERS_SEARCH_LIST = {
+            '*': ['slot1', 'slot2', etc. ], #mandatory
+            'reverse_id_alpha': ['slot1', 'slot2', etc. ],
+            'reverse_id_beta': ['slot1', 'slot2', etc. ],
+            # exclude it from the placehoders search list
+            # (however better to remove at all to exclude it)
+            'reverse_id_empty': []
+            etc.
+        }
+
+        Put '-' (minus) before the slot string to exclude it
+
+        'slot_name' -> search for slot_name
+        '-slot_name' -> exclude slot name from the search list
+
+        or leave it empty
+
+        PLACEHOLDERS_SEARCH_LIST = {}
+        """
         current_page = obj.page
-        placeholders = current_page.placeholders.all()
+        reverse_id = current_page.reverse_id
+        args = {}
+
+        try:
+            placeholders_by_page = settings.PLACEHOLDERS_SEARCH_LIST
+        except AttributeError:
+            placeholders_by_page = {}
+
+
+        if placeholders_by_page:
+            filter_target = None
+            if '*' in placeholders_by_page:
+                filter_target = '*'
+            if reverse_id and reverse_id in placeholders_by_page:
+                filter_target = reverse_id
+            if not filter_target:
+                raise AttributeError('Leave PLACEHOLDERS_SEARCH_LIST empty or set up at least the generic handling')
+            slots = [el for el in placeholders_by_page[filter_target] if not el.startswith('-')]
```

Comment:
considering ```-something``` can easily be a valid placeholder slot name, I suggest we use a different character either ```!slotname``` or maybe just use a dict with ```{'include': [], 'exclude': []}```, the dictionary approach to me seems more explicit.